working/0546-patterns/patterns-feature-specification.md

```diff
@@ -1501,17 +1538,44 @@ The context type schema for a pattern `p` is:
         doesn't destructure anything, it matches any list, so it is permissive
         with the context type.*
 
-    3.  Else `E` is the greatest lower bound of the type schemas of all element
-        subpatterns. *We use the greatest lower bound to ensure that the outer
-        collection type has a precise enough type to ensure that any typed field
-        subpatterns do not need to downcast:*
+    3.  Else, infer the type schema from the subpatterns:
 
-        ```dart
-        var [int a, num b] = [1, 2];
-        ```
+        1.  Let `es` be an empty list of types.
+
+        2.  For each subpattern `e` in `p`:
+
+            1.  If `e` is a rest element pattern with a subpattern `s` and the
+                context type schema of `s` is a `List<T>` for some type `T`,
+                then add `T` to `es`.
+
+            2.  Else if `e` is not a rest element pattern, add the context
+                type schema of `e` to `es`.
 
-        *Here, the GLB of `int` and `num` is `int`, which ensures that neither
-        `int a` nor `num b` need to downcast their respective fields.*
+            *Else, `e` is a rest element with no subpattern, so it doesn't
```

Comment:
No, `e` can be a rest element with a subpattern with some other type or no type, right?  I think this should probably not be an error: `var [...rest] = <int>[]`, but `rest` has no context type schema here.  I'm not sure whether `case [...String rest]` is intended to be an error or not - I asked about that above.

Reply:
Done (except for the `FutureOr` bit which seems a little gratuitous).